components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationResponseProcessor.java

```diff
@@ -71,12 +73,17 @@ public ActionExecutionStatus<Success> processSuccessResponse(Map<String, Object>
             ActionInvocationSuccessResponse actionInvocationSuccessResponse)
             throws ActionExecutionResponseProcessorException {
 
-        /* TODO: 1. Validate whether authenticated user data are present in the action invocation response for the
-            IDENTIFIER local authenticators and Federated authenticators.
-            2. Move this code logic to the processAuthenticationResponse method in the AbstractAuthenticatorAdapter.
-         */
         AuthenticationContext context = (AuthenticationContext) eventContext.get(
                 AuthenticatorAdapterConstants.AUTH_CONTEXT);
+        AuthenticatorPropertyConstants.AuthenticationType authType = (AuthenticatorPropertyConstants.AuthenticationType)
+                eventContext.get(AuthenticatorAdapterConstants.AUTH_TYPE);
+        if (actionInvocationSuccessResponse.getData() == null) {
+            if (AuthenticatorPropertyConstants.AuthenticationType.IDENTIFICATION.equals(authType)) {
+                throw new ActionExecutionResponseProcessorException("The authenticated user data is not found in the " +
+                        "action response from the authentication extension.");
```

Comment:
This requires diagnostic logs. Let's adda a todo and cover all diagnostics in a separate pr

Reply:
Tracking with https://github.com/wso2/product-is/issues/22753